src/schema/v2/infiniteDiscovery/resetDiscoveryArtworkReferencesMutation.ts

```diff
@@ -0,0 +1,101 @@
+import {
+  GraphQLBoolean,
+  GraphQLNonNull,
+  GraphQLObjectType,
+  GraphQLString,
+  GraphQLUnionType,
+} from "graphql"
+import { ResolverContext } from "types/graphql"
+import {
+  formatGravityError,
+  GravityMutationErrorType,
+} from "lib/gravityErrorHandler"
+import { mutationWithClientMutationId } from "graphql-relay"
+import { generateUuid } from "./discoverArtworks"
+import fetch from "node-fetch"
+
+const SuccessType = new GraphQLObjectType<any, ResolverContext>({
+  name: "DeleteDiscoveryUserReferencesMutationSuccess",
+  isTypeOf: (data) => data.success,
+  fields: () => ({
+    success: {
+      type: GraphQLBoolean,
+      resolve: (result) => result.success,
+    },
+  }),
+})
+
+const FailureType = new GraphQLObjectType<any, ResolverContext>({
+  name: "DeleteDiscoveryUserReferencesMutationFailure",
+  isTypeOf: (data) => {
+    return data._type === "GravityMutationError"
+  },
+  fields: () => ({
+    mutationError: {
+      type: GravityMutationErrorType,
+      resolve: (err) => err,
+    },
+  }),
+})
+
+const ResponseOrErrorType = new GraphQLUnionType({
+  name: "DeleteDiscoveryUserReferencesResponseOrError",
+  types: [SuccessType, FailureType],
+})
+
+export const DeleteDiscoveryUserReferencesMutation = mutationWithClientMutationId<
+  { userId: string },
+  any,
+  ResolverContext
+>({
+  name: "DeleteDiscoveryUserReferencesMutation",
+  description: "Deletes a user artwork references in weaviate",
+  inputFields: {
+    userId: {
+      description: "The user's ID",
+      type: GraphQLNonNull(GraphQLString),
+    },
+  },
+  outputFields: {
+    deleteDiscoveryUserReferencesResponseOrError: {
+      type: ResponseOrErrorType,
+      description: "On success: return boolean. On failure: MutationError.",
+      resolve: (result) => {
+        return result
+      },
+    },
+  },
+  mutateAndGetPayload: async ({ userId }, {}) => {
+    const weaviteUserId = generateUuid(userId)
+
+    try {
+      await fetch(
+        `https://weaviate.stg.artsy.systems/v1/objects/InfiniteDiscoveryUsers/${weaviteUserId}`,
+        {
+          method: "PATCH",
+          headers: {
+            "Content-Type": "application/json",
+            Accept: "application/json",
+          },
+          body: JSON.stringify({
+            properties: {
+              likedArtworks: null,
+              seenArtworks: null,
+            },
+          }),
+        }
+      )
```

Comment:
Obvious anti-pattern here of using fetch directly instead of a defined loader. Our canonical pattern would require us making changes to `metaphysics` `fetch` library, which seemed overkill and potentially risky. Specifically, it would require us to add support for the `PATCH` method (not a big deal) and we would have to change how [we parse query params](https://github.com/artsy/metaphysics/blob/main/src/lib/apis/fetch.ts#L22) to [use `strictNullHandling: true`](https://github.com/ljharb/qs/blob/d185cee45f17f1a69705d034eb370dbbbc9fba2d/README.md?plain=1#L598), otherwise, our `null` values here are converted to empty strings by `qs`, which breaks `weaviate`. Same issue if we try to pass an empty array. 

If we decide we are going to add production support for weaviate, we will need fix this.

Reply:
👍  done in [a6b6bae](https://github.com/artsy/metaphysics/pull/6211/commits/a6b6baef8083a02601bc45be7b201123fda31756)